model/project.go

```diff
@@ -1299,43 +1288,46 @@ func (p *Project) ResolvePatchVTs(bvs, tasks []string, requester, alias string,
 		}
 	}
 
-	var pairs []TVPair
+	var pairs TaskVariantPairs
 	for _, v := range bvs {
 		for _, t := range tasks {
 			if p.FindTaskForVariant(t, v) != nil {
-				pairs = append(pairs, TVPair{Variant: v, TaskName: t})
+				pairs.ExecTasks = append(pairs.ExecTasks, TVPair{Variant: v, TaskName: t})
+			} else if p.GetDisplayTask(v, t) != nil {
+				pairs.DisplayTasks = append(pairs.DisplayTasks, TVPair{Variant: v, TaskName: t})
 			}
 		}
 	}
 
 	if alias != "" {
-		aliasPairs, displayTaskPairs, err := p.BuildProjectTVPairsWithAlias(alias)
-		if err != nil {
-			grip.Error(errors.Wrap(err, "failed to get task/variant pairs for alias"))
-		} else {
-			pairs = append(pairs, aliasPairs...)
-			for _, pair := range displayTaskPairs {
-				if !utility.StringSliceContains(bvs, pair.Variant) {
-					bvs = append(bvs, pair.Variant)
-				}
-				if !utility.StringSliceContains(tasks, pair.TaskName) {
-					tasks = append(tasks, pair.TaskName)
-				}
-			}
+		catcher := grip.NewBasicCatcher()
+		vars, err := FindAliasInProject(p.Identifier, alias)
+		catcher.Add(errors.Wrap(err, "can't get alias from project"))
+
+		var aliasPairs, displayTaskPairs []TVPair
+		if !catcher.HasErrors() {
+			aliasPairs, displayTaskPairs, err = p.BuildProjectTVPairsWithAlias(vars)
+			catcher.Add(errors.Wrap(err, "failed to get task/variant pairs for alias"))
+		}
+		grip.Error(catcher.Resolve())
```

Comment:
This seems easy to get lost especially since there might be user-relevant errors in here. At the very least we should have this structured in a message.Fields

Reply:
that makes a lot of sense.
done